backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/audio/OpenALMusicProcessor.java

```diff
@@ -0,0 +1,123 @@
+package com.badlogic.gdx.backends.lwjgl3.audio;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+
+public class OpenALMusicProcessor implements Runnable, Disposable
+{
+	private Queue<OpenALMusic> musicsToPlay = new ConcurrentLinkedQueue<OpenALMusic>();
+	private Queue<OpenALMusic> musicsToStop = new ConcurrentLinkedQueue<OpenALMusic>();
+	Queue<OpenALMusic> musicsStopped = new ConcurrentLinkedQueue<OpenALMusic>();
+	
+	Queue<OpenALMusic> musicsFinished = new ConcurrentLinkedQueue<OpenALMusic>();
+	private Array<OpenALMusic> musicsToUpdate = new Array<OpenALMusic>();
+	
+	private Queue<OpenALMusic> musicsToSeek = new ConcurrentLinkedQueue<OpenALMusic>();
+	
+	private boolean shouldRun;
```

Comment:
I think shouldRun should be volatile. Otherwise, at least in theory (*), it'd be undefined if and when the thread will stop running.

* I don't think it will be a problem in pratice but since volatile shouldn't introduce much of a performance penalty, I don't see a reason to take the risk that on some systems in some weird situations the thread keeps being alive for longer or forever.